api/src/main/java/io/opencensus/stats/MeasureMap.java

```diff
@@ -89,4 +91,50 @@ public MeasureMap putAttachment(String key, String value) {
    * @since 0.8
    */
   public abstract void record(TagContext tags);
+
+  /**
+   * Records all of the measures at the same time, with the implicit current {@link TagContext} plus
+   * the additional {@link Tag}s.
+   *
+   * <p>This method records all of the stats in the {@code MeasureMap} every time it is called.
+   *
+   * <p>This is equivalent to calling:
+   *
+   * <pre>{@code
+   * TagContext context = tagger.currentBuilder().put(tagKey, tagValue).build();
+   * measureMap.record(context);
+   * }</pre>
+   *
+   * @param tags additional {@link Tag}s to be recorded against.
+   * @since 0.20
+   */
+  public void recordWithTags(List<Tag> tags) {
```

Comment:
I think here we currently support only "upsert" (replace the previous tag) but we may want to support other operations as well. So probably something like what golang did to have a notion of "TagMutator" is useful.

Also not sure if we need the metadata (the new Tag component), but happy to have it here, maybe just comment that metadata is ignored.

Before coding this idea let's make sure we agree on what to do. Thanks.